sound/soc/sof/core.c

```diff
@@ -383,6 +409,7 @@ static int sof_remove(struct platform_device *pdev)
 	snd_soc_unregister_component(&pdev->dev);
 	snd_sof_fw_unload(sdev);
 	snd_sof_ipc_free(sdev);
+	sof_virtio_miscdev_unregister();
```

Comment:
if (sdev->is_vbe) ??

Reply:
the same like upper. sof_virtio_miscdev_unregister() will do nothing if not vBE.